# Introdução à Ciência de Dados - UFPB
Professor: Yuri Malheiros

## Medidas de Centralidade e Dispersão

A partir do dataset `movies.csv` calcule a média, mediana e desvio padrão para as colunas:

### US Gross

In [2]:
import pandas as pd
import numpy as np

'''def mediana (x):
    
    serie = x[x >= 0].sort_values()
    mediana = None
    meio = int(len(serie)/2)
    
    if len(serie) % 2 == 0:
        mediana = (serie.iloc[meio] + serie.iloc[meio - 1])/2
        
    else:
        meio = int(meio + 0.5)
        mediana = serie.iloc[meio]
        
    return mediana'''
        
def mostrar_dados (x):
    
    print(f'Média = {x.mean():.2f}, Mediana = {x.median()}, Desvio padrão = {x.std():.2f}')
          
    return

filmes = pd.read_csv("movies.csv")
us_gross = filmes["US Gross"]


mostrar_dados(us_gross)

Média = 44002085.16, Mediana = 22019465.5, Desvio padrão = 62555311.39


In [3]:
us_gross.median()

22019465.5

### Production Budget

In [4]:
production_budget = filmes["Production Budget"]
mostrar_dados(production_budget)

Média = 31069171.45, Mediana = 20000000.0, Desvio padrão = 35585913.44


### Running Time

In [5]:
running_time = filmes["Running Time (min)"]
mostrar_dados(running_time)

Média = 110.19, Mediana = 107.0, Desvio padrão = 20.17


### IMDB Rating

In [6]:
imdb_rating = filmes["IMDB Rating"]
mostrar_dados(imdb_rating)

Média = 6.28, Mediana = 6.4, Desvio padrão = 1.25


### IMDB Votes

In [7]:
imdb_votes = filmes["IMDB Votes"]
mostrar_dados(imdb_votes)

Média = 29908.64, Mediana = 15106.0, Desvio padrão = 44937.58


### Análise dos Resultados

Quais colunas tiveram valores muito diferentes entre média e mediana? Quais os seus desvios padrão?

US Gross: Desvio → 62555311.39

Production Budget: Desvio → 35585913.44

IMBD Votes: Desvio → 44937.58

**RESPONDA AQUI**

Que conclusões podemos tirar desses resultados?

Baseado na discrepância entre a mediana e a média dessas colunas, é possível inferir que existe uma distribuição distorcida e alguns valores influenciam bastante na média e, pelo desvio padrão, é notável o quão disperso esses valores estão.

Para os outros valores, é observado que a mediana e a média são extremamente próximas e o desvio mostra que os valores não diferem muito da média.

**RESPONDA AQUI**